3.0.0 changes

cs-abhishek-shukla opened the pull request:
- Added minimum permissions table
- Added release notes
- added sections in usage
- Removed redundant info
- Uploaded the new instruction set

cs-suyog-jinturkar commented on docs/res/FortiAIPBGeneration.png:
Please retain old image and do not add new as we are not promoting playbook generation from any page other than PB designer.

Please add a known issue: Playbook generated using FortiAI bot on any page other than PB designer will not be copied in a clipboard. Use PB designer to generate a PB using FortiAI bot

cs-abhishek-shukla replied:
Added instruction to open playbook designer to create playbook

cs-suyog-jinturkar commented on docs/res/SOC-Assistant-Instruction.txt:
```diff
@@ -0,0 +1,85 @@
+Role Description: 
+You are a expert and helpful Cybersecurity SOC assistant/analyst called FortiAI, responsible for monitoring, detecting, and responding to cybersecurity incidents. Your primary duties include analyzing security events, investigating potential threats, and implementing measures to protect the organization’s information systems. You will utilize various security tools and threat intelligence to safe ...
```
Why this file is added here? Please remove it

cs-suyog-jinturkar replied:
We are not going to share or expose this instruction set to the user.

cs-abhishek-shukla replied:
Ok. Removed.

cs-suyog-jinturkar commented on docs/res/instruction-update-00.png:
please remove this SS as well as any steps related to updating instruction. We are not going to expose this functionality to the user. This is recently communicated to us hence the change.

cs-abhishek-shukla replied:
Removed the section.

cs-suyog-jinturkar commented on docs/res/instruction-update-01.png:
same as above.

cs-abhishek-shukla replied:
Removed the section

cs-suyog-jinturkar commented on docs/res/instruction-update-02.png:
same as above.

cs-abhishek-shukla replied:
Removed the section

cs-suyog-jinturkar commented on docs/res/llm-configuration.json:
```diff
@@ -0,0 +1,22 @@
+        "title": "OpenAI",
+        "modelList": [
+          "gpt-3.5-turbo",
+          "gpt-3.5-turbo-0301",
```
this is old data from previous release. please use latest

cs-suyog-jinturkar replied:
@cs-abhishek-shukla  please do the needful

cs-abhishek-shukla replied:
Changed

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
-- You have An OpenAI account and a valid API key to access the OpenAI APIs.
-- *GPT4*, i.e. the `gpt-4-1106-preview` model is enabled for use in your OpenAI account. Versions prior of *GPT4* do not generate correct responses. 
+- An OpenAI account and a valid API key to access the OpenAI APIs.
+- Access to *GPT4*, i.e. the `gpt-4-1106-preview` model. Versions prior to *`GPT4`* do not generate correct responses.
```
correct --> as expected

cs-abhishek-shukla replied:
Changed

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
 
-- Read and Usage permissions on Widgets
-- Read and Execute permissions on Playbooks
+The following permission allow an administrator to run the [configuration wizard](#fortiai-configuration-wizard) and update the instruction set:
```
and update the instruction set: --> please remove this. Please keep instruction data with you but do not add here

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
+| Widgets        | ![No](./res/icon-close.svg)        | ![Yes](./res/icon-green-check.svg) | ![Yes](./res/icon-green-check.svg) | Usage                       |
+
+>[!Note]
+> The AI widget becomes available to interact only on modules that contain records. For example: Alerts, Indicators, or Incidents.
```
1. AI widget --> FortiAI bot
2. FortiAI bot will be available on modules even if there are no records present.

cs-abhishek-shukla replied:
Changed

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
+    ![](./res/config-wizard-01.png)
+
+    - **LLM Integration** &mdash; *OpenAI* (default).
+    - **Conversation Model**: Select one from the following options:
```
please mention that these values are just for representational purposes only.

cs-abhishek-shukla replied:
But these are the only options available

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
+- Click **Finish** to complete the configuration.
 
-    ![FortiAI configuration get started](./res/config-wizard-01.png)
+Navigate to *Incident Response* or *Resources* navigation menu to interact with **FortiAI**.
```
we should refer either FortiAI or Advisor in the doc

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
 
-3. Specify LLM Integration to use and related model for Conversation, Playbook Generation mode and click **Next**.
+> [!IMPORTANT]  
+> By default, all these fields are loaded from and overwritten to the **Key Store** record *`fortiai-configuration`*.
```
overwritten or saved once wizard finishes

cs-abhishek-shukla replied:
changed

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -12,57 +12,122 @@
-4. Click **Finish** to complete the configuration process.
+1. Generate a Project Level API key in OpenAI. For information refer to [API Keys](https://platform.openai.com/docs/api-reference/api-keys) section of the OpenAI documentation.
+2. Run the FortiAI Config wizard. Refer to the section [FortiAI Configuration Wizard](#fortiai-configuration-wizard) for information on launching the wizard.
+3. Give the project name in wiz
```
wiz?

cs-suyog-jinturkar replied:
give project id and not name in openai connector config page

cs-abhishek-shukla replied:
Removed this section

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
-Type your query, such as
+1. Escalate an alert to incident
+2. Escalate multiple alerts to Incident
+3. Rename the existing assistant
```
Rename the existing assistant --> we don't support this

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
+1. Escalate an alert to incident
+2. Escalate multiple alerts to Incident
+3. Rename the existing assistant
+4. Add and remove values in the exclude list
```
we don't support this

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
+2. Escalate multiple alerts to Incident
+3. Rename the existing assistant
+4. Add and remove values in the exclude list
+5. Investigating an Alert: For an Unknown Type Alert ingested
```
supported for any type hence no need to mention this.

cs-suyog-jinturkar replied:
There are multiple other actions like navigation, create warroom, escalate an alert, block IOC, enrich IOC, create/update record, etc. are also supported. Complete list is already given to you by Ankit on the chat.

Bot also helps user to create jinja or regular expressions which can be hndy while developing playbooks. Apart from this, bot is very helpful get additional information about threat attacks, treat actors, CVE, etc. (instructions can be helpful here to form an entire section)

cs-suyog-jinturkar replied:
You also form a step by step use-case starting with the alert ingested, extracted, enriched and then soc analyst opened the bot to start investigation, gain more information about alert, generated response plan, gather miter insights, get ioc enrichment using bot, etc. If found malicious and other details supports then escalate this alert to an incident, update some fields like phase, etc. Ask bot to do summary, create recommendations. and with this you can block ioc and close the incident or create a war-room for further invetigtion.

cs-abhishek-shukla replied:
Removed the section

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
+You may not be a seasoned playbook developer yet and may require some assistance with developing this playbook.
 
-Typing your requirements, such as
+Use FortiAI to help with this task by clicking the **Playbook Generation** option in the FortiAI flyout.
```
this option is no longer available

cs-abhishek-shukla replied:
Fixed

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
+![FortiAI - Suggested playbook outline](../docs/res/FortiAIPBGeneration.png)
 
-Assume for the moment that you are a playbook designer who is required to build workflows to mitigate various types of threats. 
+You can modify the playbook outline presented, such as changing the integration from VirusTotal to IPStack. Once you are satisfied, click **Ok, Generate Playbook**:
```
modify using UI or you can use prompts and bot can do it for you like add set variable step to store result returned by virustotal

cs-suyog-jinturkar replied:
try this using bot then you can write it better.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
+Consider a situation where you are assigned to create a plan of action for investigating a ransomware event.
 
-![FortiAI - Suggested playbook outline](../docs/res/FortiAIPBGeneration.png)
+You can use *FortiAI* to help with this task by opening a playbook in the playbook designer and clicking the **Response Plans / Jinja / Playbook How To's** option in the FortiAI flyout.
```
clicking the **Response Plans / Jinja / Playbook How To's** option --> remove now

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
-    <th>NOTE</th>
-    <td>If any step uses a connector that has not been installed, then FortiAI will display a message informing the user that the specific step requires a connector that is not installed on the system.</td>
-</table>
+Click the **Post** icon. FortiAI, in response, provides you with a good framework for a response plan that you can use to design playbooks utilizing the **Playbook Generation** option:  
```
this option is not present now, its removed.

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,79 @@
+You might need assistance with Jinja expressions found in reference playbooks, or you might need specific Jinja values. For instance, you might be looking for a Jinja to trim HTML from a string input.
 
-Post-review, you can connect the AI generated playbook blocks to the main playbook flow and create a complete workflow.
+Use FortiAI to help with this task by clicking the **Response Plans / Jinja / Playbook How To's** option in the FortiAI flyout.
```
**Response Plans / Jinja / Playbook How To's** option --> removed. user still can use NLP and get the desired results.

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/usage.md:
please remove instruction update part from the doc

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/usage.md:
please remove filter record point, as it was not merged due to regression impact.

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/usage.md:
heading could be -- supported use-cases using fortisoar actions or FortiAI supported usecases and functions or etc

cs-abhishek-shukla replied:
Removed heading

cs-suyog-jinturkar commented on docs/usage.md:
no need to document this as bot does this internally.

cs-abhishek-shukla replied:
Edited.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,41 +1,31 @@
+
+- The playbook collection *10 - SP - Fortinet Advisor* has been renamed to **10 - SP - FortiAI**.
+
+- The following playbooks have been deprecated:
```
note can be added here, these actions are now possible in bot using openai assistant without triggering any playbook.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -1,41 +1,31 @@
-
-- Specific modules now have pre-defined questions. For example, under the **Alerts** module, questions include:
-
+- **FortiAI** solution pack has an elevated ability to understand and execute following tasks directly as part of conversations, transforming complex security operations into straightforward dialogue:
```
RN should also talk about how easy it is now to do conversation with bot and perform the soc tasks seamlessly since it has record context and it take consideration of previous conversations .

cs-suyog-jinturkar replied:
Also, all the FortiSOAR actions like creating or modifying reocrds, triggering playbooks happens in logged-in users context. Hence, bot will honour FortiSOAR users RBAC settings while performing any action.

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on release_notes.md:
```diff
@@ -1,41 +1,31 @@
-    - **Conversation Mode**: gpt-3.5-turbo
+>[!NOTE]
+>Manually delete these playbooks if upgrading from an older **FortiAI** version.
 
```
1. Added new attachment file to store SOC Assistant Instruction
2. OpenAI new model support (copy it from FortiAI Config Wizard)

cs-abhishek-shukla replied:
done

cs-suyog-jinturkar commented on docs/advanced-usage.md:
```diff
@@ -5,7 +5,7 @@
 ## Prompting tips
 
-Simple tweaks to the input prompts could improve the playbook block generation process using FortiAI. While asking queries or giving input to the FortiAI keep some tips in mind:
+Simple tweaks to the input prompts could improve the playbook block generation process using **FortiAI**. While asking queries or giving input to the Advisor keep some tips in mind:
```
Advisor --> Assistant bot

cs-abhishek-shukla replied:
changed

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
                 <tr>
-                    <th>modelList</th>
+                    <th><code>modelList</code></th>
                     <td>List of LLM Integration models. By default, `gpt-3.5-turbo`, `gpt-3.5-turbo-0301`, `gpt-4`, `gpt-4-1106-preview` is present.</td>
```
these values needs to be updated as latest mentioned in config wizard

cs-abhishek-shukla replied:
Updated

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
-        <th>llmIntegrations</th>
+        <th rowspan=3><code>fortiai-configurations</code></th>
+        <th><code>llmIntegrations</code></th>
         <td>Sets the integration such as OpenAI, Google Bard, etc., to be used as the LLM. By default, 'OpenAI' is configured to be used as the LLM.
```
remove Google Bard please

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
     <tr>
-        <th>llmIntegrationToUse</th>
+        <th><code>llmIntegrationToUse</code></th>
         <td>Sets the integration such as OpenAI, Google Bard, etc., to be used as the LLM. By default, `OpenAI` is configured to be used as the LLM.</td>
```
please remove Google Bard

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
                 <tr>
-                    <th>pBGenerationModel</th>
+                    <th><code>pBGenerationModel</code></th>
                     <td>Sets the model of LLM Integration to use in <i>Playbook Generation</i> Mode. By Default, `gpt-4-1106-preview` is to be used. </td>
```
By Default, `gpt-4-1106-preview` --> needs to be replaed with latest

cs-abhishek-shukla replied:
Updated

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
-    </tr>
-</table>
+>[!NOTE]
+>To add a deprecated question for different modules, follow these steps:
```
deprecated --> what does this means here?

cs-abhishek-shukla replied:
Removed

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
+>[!NOTE]
+>To add a deprecated question for different modules, follow these steps:
+>   1. Add the question in JSON format under the particular module's API name. A sample [Question JSON](./question-sample.json) is attached to the document.
+>   2. Add new LLM or LLM models in [](./),Write a playbook along with the tag mentioned in the JSON above.
```
No need to write a PB now onwards. @cs-ankit-gupta please review this section carefully.

cs-ankit-gupta replied:
Now user don't need to add PB

cs-abhishek-shukla replied:
Removed line `,Write a playbook along with the tag mentioned in the JSON above.`

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
+
+| Playbook Name                                                                  | Description                                                                               |
+|:-------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------|
+| Get Playbook Step Suggestion                                                   | Suggests steps to create a playbook based on user's query.                                |
```
Many of these playbooks are removed. Please verify once and make required changes.

cs-suyog-jinturkar replied:
@cs-ankit-gupta  for your notice

cs-abhishek-shukla replied:
Marked deprecated

cs-suyog-jinturkar commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
+
+
+>[!Warning]
+>We recommend that you clone these playbooks before customizing to avoid loss of information while upgrading the solution pack.
```
No needed anymore

cs-abhishek-shukla replied:
Leaving as-is

cs-suyog-jinturkar commented on docs/res/llm-configuration.json:
```diff
@@ -0,0 +1,22 @@
+    ],
+    "llmIntegrationData": {
+      "openai": {
+        "conversationModel": "gpt-4-1106-preview",
```
same as above

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on README.md:
```diff
@@ -1,31 +1,21 @@
 # Overview
 
-FortiSOAR, in its unwavering quest for efficiency and user-centricity, reaches another significant milestone with the release of its AI-powered security assistant, the "FortiAI".
+Using the power of generative AI, the **FortiAI** solution pack answers your questions contextually regarding security threats, response processes, work plans, [Jinja](https://jinja.palletsprojects.com/en/3.1.x/) expressions, and more. Its unique feature is its ability to generate playbook blocks tailored to scenarios defined by you, giving you a head start in automating and enhancing your use ca ...
```
We need to add more about SOC Investigation and how fortiAi make it easier for you

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on README.md:
```diff
@@ -1,31 +1,21 @@
-### Data Privacy Notice 
-
-The FortiAI lets you connect your Fortinet account to your own OpenAI account, using your own OpenAI ChatGPT API key. This integration will send data from your Fortinet account to OpenAI and will show you responses from OpenAI. Fortinet does not verify or correct these responses and has no responsibility for them.  ChatGPT is operated by a third party, not Fortinet. You must exercise discretion a ...
+The FortiAI lets you connect your Fortinet account to your own OpenAI account, using your own OpenAI ChatGPT API key. This integration will send data from your Fortinet account to OpenAI and will show you responses from OpenAI. Fortinet does not verify or correct these responses and has no responsibility for them. ChatGPT is operated by a third party, not Fortinet. You must exercise discretion an ...
```
OpenAI ChatGPT API Key - OpenAI Assistant API Key

cs-abhishek-shukla replied:
done

cs-ankit-gupta commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
-| Name                     | Description                                                                                                                                       |
-|:-------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------|
-| FortiAI - SOC Assistant Instructions   | Contains the set of instruction of which SOC Assistant will get created.            |
+| Name                                 | Description                                                              |
```
Duplication of Attachment record

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
 
```
AI Assistant Utils - It also communicate with respective LLM like OpenAI

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
+| Name                                                        | Description                                                                                          |
+|:------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------|
+| AI Assistant                                                | Adds the Advisor bot to the FortiSOAR User Interface.                                                |
+| FortiAI Configuration Wizard![icon-new](./res/icon-new.svg) | Launches a wizard that walks a user through the process of setting up LLM Integration Configuration. |
```
Remove new image

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -37,3 +37,16 @@
+
+    - The bot now has increased access to various modules, maintaining privacy is paramount. Sensitive data is still being actively masked for all interactions with the LLM.
+
+    - The bot can now hold conversations to understand requirements and generate a more targeted playbook.
```
playbook generation experience has been streamlined with the introduction of the OpenAI assistant. Users can now engage with the bot during the creation of a playbook outline, making it easier to modify the outline by adding, updating, or removing steps.

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on docs/contents.md:
```diff
@@ -14,134 +14,143 @@ The **FortiAI** solution pack contains the following resources:
         </td>
     </tr>    
 </table>
 
```
New note here - To include correlated filed of record e.g., from alert record, if you want to add the correlated assets records as well so add new module name as { "assets": "fieldsOfInterest": ["id", "@id", "hostname"]}

cs-abhishek-shukla replied:
Done in advanced

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -37,3 +37,16 @@
+
+    - The bot can now hold conversations to understand requirements and generate a more targeted playbook.
+
+    - Through enhanced comprehension, the bot can also navigate to modules, its records, and playbook logs; in addition to alert investigation, remediation, escalation to incident, and even setting up war rooms, among several others.
```
in addition to alert investigation, remediation, --> these are some soc use-cases are introduced in this release hence needs more attention.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/llm-configuration.json:
```diff
@@ -4,19 +4,20 @@
-        "conversationModel": "gpt-4-1106-preview",
-        "pBGenerationModel": "gpt-4-1106-preview",
-        "isMultiConfigAvailable": false
+        "messagesLimit": 20,
```
WE need to explain this value in the advance-usage section

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/advanced-usage.md:
```diff
@@ -143,7 +141,5 @@ To resolve this issue, try the following:
 - Rephrase the prompt and try to generate the playbook steps, see [Prompting tips](#prompting-tips).
 - Verify that your OpenAI account is operational and has enough credit.
```
Need to add details about "messagesLimit" settings from key-store

"llmIntegrationData": {
      "openai": {
        "messagesLimit": 20,

cs-abhishek-shukla replied:
mentioned in Contents

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
+8. **Update Record**: Update a record using the updated information provided.
+9. **Escalate To Incident**: Escalate, elevate, or promote the alert to an incident.
+10. **Setup War Room**: Establish, set up, initiate, or create a war room for an incident.
+11. **Filter Record**: Filter module records using a specified condition or filter.
```
need to remove this point.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
+1. **Generate Playbook**: Generate, create a playbook or workflow, or playbook steps.
+2. **Navigate Pages**: Navigate, open, or move to a specific page or module.
+3. **Provide Context**: Load, retrieve, fetch, access, or provide more information about an *already opened* record.
+4. **Link Record**: Link or correlate a record with other relevant records. For example: correlating an indicator to an existing alert.
```
Need to add this point (#4) in the release-notes

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
+3. **Access Detailed Records**:
+   - Access detailed information about the record as needed. For example:
+
+    >*Access the full details of the alert for further analysis*
```
Need to change this prompt @cs-ankit-gupta

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
+
+A common use case involves providing additional details about an alert or incident that is already opened. For example, an analyst might need to access the full context of an alert to understand its background and relevance. By providing additional context, the analyst can make more informed decisions and take appropriate actions.
+
+### Providing Context Using FortiAI Assistant
```
Do we really nee this section? @cs-ankit-gupta

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
+
+- Review the additional information provided to ensure it meets your needs. Verify that the context is accurate and relevant to the current task or analysis.
+
+### Integrate Context into Workflow
```
Remove this section as this is not needed

cs-ankit-gupta commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
-Consider a situation where you are assigned to create a plan of action for investigating a ransomware event.
+1. **Generate Playbook**: Generate, create a playbook or workflow, or playbook steps.
+2. **Navigate Pages**: Navigate, open, or move to a specific page or module.
+3. **Provide Context**: Load, retrieve, fetch, access, or provide more information about an *already opened* record.
```
We can remove this point, as this won't be frequently by user

cs-abhishek-shukla replied:
removed

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,317 @@
+   - Specify the records you want to correlate. For example:
 
-Based on the reputation, the indicator reputation is updated to *Malicious*, if the reputation returned by VirusTotal is *Suspicious*; or the indicator reputation is updated to *Good* if the reputation is anything but *Suspicious*.
+    >*Correlate the newly detected IP address with existing incidents*
```
Change this prompt to "Correlate the indicator id 1 address with this incidents"

cs-abhishek-shukla replied:
Done

cs-abhishek-shukla committed:
3.0.0 changes
- Added minimum permissions table
- Added release notes
- added sections in usage
- Removed redundant info
- Reviewed and updated setup.md
- Added SS as part of review
- usage.md changes
    - Generate Playbook
    - Navigate Pages
    - Link Record
    - Create Record
    - Update Record
    - Block Indicator
    - Enrich Indicator
    - Escalate To Incident
    - Setup War Room

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+    >*Open the record with ID [ALERT-ID]*
+
+   - The assistant locates and displays the record for you.
```
Add below point for navigation, 
It also gives you a hyper link for the said record in response. It is useful when user refers to past conversation and still able to navigate using these link present in the bot conversation.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+Adjust the navigation command to suit your needs, such as specifying a different module or record.
+
+### Review Navigation Commands
```
What is a use of "Review Navigation Commands" section?

cs-abhishek-shukla replied:
For consistency

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+### Use Case Example: Correlating an Indicator
+
+A common use case is correlating an indicator with an existing alert. For example, if an alert is triggered due to a specific indicator, linking the indicator to the alert helps in understanding the relationship and impact. This correlation can aid in more accurate threat analysis and response.
```
For example, if an alert is triggered due to a specific indicator, --> this is wrong. Why would alert will be triggered due to a specific indicator?

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+2. **Correlate Records**:
+   - Specify the records you want to correlate. For example:
+
+    >*Correlate the newly detected IP [IP-ADDRESS] with the [INCIDENT-ID]*
```
as discussed, remove this example, as we don't co-relate using value. user need to provide id or uuid. Which covers in above example.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+3. **Review Linked Records**:
+   - Review the links and correlations to ensure they are correct and relevant. Verify that the connections between records provide meaningful insights.
+
+### Review Linked Records
```
IMO, there is no need of this "Review Linked Records" section, please remove it

cs-abhishek-shukla replied:
For consistency

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+- Utilize the linked records to enhance your workflow. The connections between records can provide valuable insights and help streamline data analysis and response actions.
+
+### Conclusion
```
Please remove this section

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+- **Setup War Room**: Establish, set up, initiate, or create a war room for an incident.
 
-You can use *FortiAI* to help with this task by opening a playbook in the playbook designer and clicking the **Response Plans / Jinja / Playbook How To's** option in the FortiAI flyout.
+>[!NOTE]
```
Move this note to below where we are actually explaining create, update, link usecases

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+   - Connect the reviewed AI-generated playbook blocks to the main playbook flow to create a complete workflow.
+
+The ability to generate playbooks through OpenAI's assistant represents a significant advancement in playbook design, offering immense assistance to those new to developing playbooks. By following the above steps, administrators can efficiently create and manage robust playbooks to address various threat mitigation scenarios.
+
```
Added below tip;

>[!TIP]
>It is recommended to clear a conversation whenever user is intended to execute different usecases. It will also help in save tokens being consumed.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+   - The assistant generates the new record based on the provided details. If any mandatory details are missed, the bot requests for the missing information.
+
+3. **Review and Confirm**:
+   - Review the details of the newly created record to ensure accuracy. Confirm the creation to finalize the record entry.
```
Confirm the creation to finalize the record entry. --> this is not needed but you can add that created record navigation link was presented to the user for easy navigation as part of response.

cs-abhishek-shukla replied:
For consistency

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+3. **Review and Confirm**:
+   - Review the details of the newly created record to ensure accuracy. Confirm the creation to finalize the record entry.
+
+### Review Created Records
```
multiple review sections, please remove this --> ### Review Created Records

cs-abhishek-shukla replied:
for consistency

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+### Review Created Records
+
+- Verify the accuracy and completeness of the created records. Ensure that all necessary information is included and that the record meets the required specifications.
```
please remove this text as well.

cs-abhishek-shukla replied:
for consistency

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+Creating records is a fundamental task for managing and tracking data within FortiSOAR. By following the steps outlined and leveraging the FortiAI assistant’s capabilities, users can effectively create and manage records to support their workflow and data management needs.
+
+>[!NOTE]
```
This note is not needed

cs-abhishek-shukla replied:
Done

cs-ankit-gupta commented on docs/advanced-usage.md:
```diff
@@ -143,7 +141,5 @@ To resolve this issue, try the following:
 - Rephrase the prompt and try to generate the playbook steps, see [Prompting tips](#prompting-tips).
 - Verify that your OpenAI account is operational and has enough credit.
 
```
Add a new file for troubleshooting and add following details
## Troubleshooting
### FortiAI not able to create or update record as per specified fields in FortiSOAR

FortiAI not able to create or update record as per specified fields in FortiSOAR

**Resolution**

- Prompt the FortiAI as - What parameter did you pass to create/update field.
- Correct the parameter as per FortiSOAR

### OpenAI Connector Configuration of FortiAI

On Changing the config name of OpenAI Connector, press Back button and then Next

### Not expected output as per prompt

On giving the prompt, if you are not getting the correct response

**Resolution**

- Again give the prompt

### Resposne delay

On giving the prompt, it is visible on disabled text input field and if loader gif is on screen.

**Resolution**

- Due to some error prompt has not been passed to LLM, refresh the page and try again with same promopt

### OpenAI Error Messages

1. Can't add messages to thread_123ndasda341 while a run run_134314hh1 is active.
**Resolution**
- Try to cancel the RUN on OpenAI platform - while loading the same thread on assistant using URL - https://platform.openai.com/playground/assistants?mode=assistant&assistant=asst_id&thread=thread_id and cancel the run
- OR Clear the Conversation

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+   - Review the updated details to ensure accuracy. Confirm the update to finalize the changes to the record.
 
->*Extract Indicators from the Alert Source Data and enrich them using VirusTotal. Based upon reputation, If `malicious` update the indicator reputation to `malicious` else if `suspicious` update the indicator reputation to `suspicious` else update indicator reputation to good*
+### Review Updated Records
```
remove multiple "Review Updated Records" sections

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
 
-![FortiAI - Generate Playbook button](../docs/res/FortiAIPBGenerationReview.png)
+>[!TIP]
+>Regularly review and update records to ensure that all information remains current and accurate. This practice helps maintain the reliability of your data and supports effective decision-making.
```
this TIP is also talk the same thing about review

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
-    <th>NOTE</th>
-    <td>If any step uses a connector that has not been installed, then FortiAI will display a message informing the user that the specific step requires a connector that is not installed on the system.</td>
-</table>
+Updating records is a crucial task for maintaining accurate and relevant information within FortiSOAR. By following the outlined steps and using the FortiAI assistant’s capabilities, users can effectively manage and update records to support their workflow and data accuracy.
```
this para is also talks the same thing about review. we don't need so many review sections.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+   - The assistant applies the changes to the record based on the provided information.
 
-Type your requirements, such as
+3. **Review and Confirm**:
```
Add a note here that updating any text type of field will result in overwriting exiting field value.

cs-abhishek-shukla replied:
Added

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
-- On Delete of an alert, using Exchange send an email to Admin
-- Find the reputation of the indicator using VirusTotal and prompt the user for action: either block it using FortiGate or take no action.
-- Manually trigger a playbook for an Alert Record to change the status to Investigating.
+    >*Block IP address `[IP-ADDRESS]` using the [configured connector]*
```
change the prompt to "">*Block IP address `[IP-ADDRESS]` *

cs-suyog-jinturkar replied:
Please also add that Block indicator playbook will be triggered to block an IOC.
Note: If playbook takes time to execute then it will present you a playbook execution log as part of response so that user can navigate and check its status.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+   - The assistant processes the request and blocks the indicator accordingly.
+
+3. **Review Blocking Status**:
+   - Verify that the indicator has been successfully blocked. Check the status or confirmation provided by the assistant to ensure the action was completed.
```
Please also add that Block indicator playbook will be triggered to block an IOC.
Note: If playbook takes time to execute then it will present you a playbook execution log as part of response so that user can navigate and check its status.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+- Confirm that the blocking action has been executed correctly and that the indicator is no longer active. Ensure that the blocking process has addressed the threat effectively.
+
+### Integrate Blocking Actions into Workflow
```
Again review and Integrate Blocking Actions into Workflow are repeated. Please remove them.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+3. **Review Incident Creation**:
+   - Verify that the incident has been created and that all relevant details from the alert have been transferred. Ensure that the incident record accurately reflects the information from the alert.
+
```
Add a note: 
1. On prompt submit, the escalate playbook will be triggered and will present manual input. 
2. If any record is open while escalating an alert then manual input will be presented on that record else it will be notified at "Pending Tasks" besides "Settings" in FortiSOAR tool bar.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+- Confirm that the escalation process has been completed successfully and that the new incident record contains all necessary information. Review the incident to ensure it is properly documented and ready for further action.
+
+### Integrate Escalated Incidents into Workflow
```
same as above. Do we really need these review sections and multiple texts?

cs-abhishek-shukla replied:
Consistency

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+   - The assistant processes the request and initiate the war room setup accordingly.
+
+3. **Review War Room Creation**:
+   - Verify that the war room has been successfully created and that all relevant details are included. Ensure that the war room is properly configured to support incident response activities.
```
Add a note:
1. On prompt submit, the setup war room playbook will be triggered and will present manual input.
2. If any record is open while escalating an alert then manual input will be presented on that record else it will be notified at "Pending Tasks" besides "Settings" in FortiSOAR tool bar.

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,130 +3,441 @@
+
+- Confirm that the war room setup process has been completed successfully and that the new war room contains all necessary information. Review the war room configuration to ensure it is ready for use.
+
+### Integrate War Room into Workflow
```
same as above. Do we really need these review sections and multiple texts?

cs-abhishek-shukla replied:
consistency

cs-abhishek-shukla committed:
3.0.0 changes
- Added Key store record information
- Removed  repetitions

cs-suyog-jinturkar commented on docs/usage.md:
```diff
@@ -3,132 +3,488 @@
+
+Creating records is a fundamental task for managing and tracking data within FortiSOAR. By following the steps outlined and leveraging the FortiAI assistant’s capabilities, users can effectively create and manage records to support their workflow and data management needs.
+
+>[!NOTE]
```
Add this text in note for create and update record section.
On MSSP setup, user will not be able to create record with a prompt containing tenant name like "Create a new alert for tenant mytenant"

cs-abhishek-shukla replied:
Done

cs-abhishek-shukla committed:
3.0.0 changes
- Added Key store record information
- Updated advanced usage doc with troubleshooting information
- Incorporated review comments
